### ISSUE TITLE
Resize all windows to fullscreen size when in fullscreen

### ISSUE THREAD

**Issue by Dickby** [hyprwm/Hyprland#683]
Would it be possible to size all windows on the active workspace to the full screen size?
What i mean is that the non visible windows are scaled anyways.
That way the movefocus animation on a fullscreen window, would look much cleaner,
i think that the animation when leaving fullscreen would look nice too, kind of spreading all windows out.

**Comment by vaxerski** [comment]
A lot of traction over time.

The question is how to approach this. Sizing all windows to fullscreen would introuce a few problems:
a) toggling fullscreen, while the other apps fade out, would still visibly animate them to the fullscreen size. Would look weird.
b) what about floating windows?
c) what about windows that _cannot_ be resized (and might corrupt when we force them to)? Currently yes, if you forcefully fullscreen one it will be fullscreen'd, but if you don't, it won't be.

**Comment by itshog** [comment]
> A lot of traction over time.
> 
> The question is how to approach this. Sizing all windows to fullscreen would introuce a few problems: a) toggling fullscreen, while the other apps fade out, would still visibly animate them to the fullscreen size. Would look weird. b) what about floating windows? c) what about windows that _cannot_ be resized (and might corrupt when we force them to)? Currently yes, if you forcefully fullscreen one it will be fullscreen'd, but if you don't, it won't be. d) some windows might visibly change their settings (like e.g. width of a sidebar, or something) when resized. You fullscreen a window A, that also changes the size of window B, then you unfullscreen A and notice B's contents are wonky. e) we cannot send the fullscreen state to every window. So, we'd only resize, and then send fullscreen events to the current window. What if that forces some changes in the contents? People would still complain about some "flashing"

I understand your points, but as far as I understand they applies only when there is an animation. Would it be possible to fix the resizing/flashing when animations are disabled? I think that many users (me included) could live without animations, but, at least for me, having some kind of flashing every time I switch between maximized windows is kind of disruptive.

Anyway, thanks for the great project!

**Comment by alex-courtis** [comment]
> I think that many users (me included) could live without animations

I do not require animations.

Hyprland is an incredible compositor however this issue is preventing my adoption.

**Comment by jovanmaric** [comment]
Yeah I agree, resizing  them eagerly would solve the flashing issues for me. Even if the app needs to re-render its contents, then it would at least be persisted while fullscreen is active. Currently the back and forth resizing sometimes bugs chrome (and possibly all electron apps and other) where it takes a second or two, before it properly resizes it's content, while the hyperland window is fullscreen.

**Comment by adamnejm** [comment]
Can we at least change the title of this thread? As you can see by the threads marked and closed as a duplicate form of this one, it is not simply a matter of *"animation enhancement"*, but a rather severe issue interrupting people's workflow and causing plethora of glitches, not only visual ones.  
  
Maybe then it would be easier to find for somebody willing to tackle this.

**Comment by weijarz** [comment]
I want toggle fullscreen like adding all current workspace windows to a group (like use togglegroup cmd), that seems doesn't have resize flash, but I don't want the top tab-bar for the group, I want to be able to display a list of the windows in the fullscreen group on the waybar.

**Comment by vaxerski** [comment]
no work is being done wrt. this. If you want to, you can pick it up yourself.

**Comment by vaxerski** [comment]
Hello there!

This issue has been closed, as we are moving from issues to discussions. More about this can be found [here](https://wiki.hyprland.org/Contributing-and-Debugging/Issue-Guidelines/) and in #9854.

Firstly, please make sure this issue is still relevant in latest Hyprland. If so, we ask you to open a [discussion](https://github.com/hyprwm/Hyprland/discussions) (please read the discussion guidelines first, in the pinned post)

Thank you for your understanding!
